### PR TITLE
Add XHTMLDocHandler class unit test

### PR DESCRIPTION
**Pull request by dwikler**
- Add unit test cases for each method in html_doc_handler.py
- Improve docstrings in xhtml_doc_handler.py

## Summary by Sourcery

Refactor XHTMLDocHandler to improve modularity and documentation, and add extensive unit tests to validate its download, caching, and parsing functionality.

Enhancements:
- Refactor download method by extracting directory creation, URL fetching, and content saving into _ensure_dir_exists, _fetch_url_content, and _save_content helpers
- Improve and enrich docstrings for XHTMLDocHandler and its methods, adding detailed Args, Returns, and Raises sections
- Add exception chaining for clearer error propagation in download and parse_dom

Tests:
- Add comprehensive unit tests for _fetch_url_content, _ensure_dir_exists, _save_content, download, parse_dom, and get_dom covering success, failure, and edge cases

### CODE REVIEW

File: src/dcmspec/xhtml_doc_handler.py [hunk 1]
```diff
@@ -28,57 +39,73 @@ def get_dom(self, cache_file_name: str, url: Optional[str] = None, force_downloa
...
+
+    def _fetch_url_content(self, url: str) -> str:
+        """Fetch content from a URL and return as UTF-8 text."""
+        response = requests.get(url, timeout=30)
+        response.raise_for_status()
+        # Force UTF-8 decoding to avoid getting Ã (/u00C3) characters
+        response.encoding = "utf-8"
+        # html_content = response.content.decode("utf-8", errors="ignore")
+        # return html_content
+        return response.text
```
[the remark this answers is not in the document]
Comment (dwikler) [reply src/dcmspec/xhtml_doc_handler.py:87]: I want to keep it for the future if needed. I added a comment to explain.

File: src/dcmspec/xhtml_doc_handler.py [hunk 2]
```diff
@@ -7,18 +13,23 @@
 
 
 class XHTMLDocHandler(DocHandler):
+    """Handler class for DICOM specifications documents in XHTML format.
+
+    Provides methods to download, cache, and parse XHTML documents, returning a BeautifulSoup DOM object.
+    Inherits configuration and logging from DocHandler.
+    """
 
     def get_dom(self, cache_file_name: str, url: Optional[str] = None, force_download: bool = False) -> BeautifulSoup:
```
[the remark this answers is not in the document]
Comment (dwikler) [reply src/dcmspec/xhtml_doc_handler.py:22]: Modified to always check url present when needed.
